src/GT/DB/Usage.hs

```diff
@@ -0,0 +1,47 @@
+-- | A 'Usage' links a 'Component' with a 'Ride'. This link is what
+-- provides the information about distance on a per-component basis.
+module GT.DB.Usage
+    ( module GT.DB.Usage
+    , module GT.DB.Schema.Usage
+    ) where
+
+import GT.DB.Prelude
+
+import Database.Persist.Sql.Raw.QQ
+import Database.Persist.Sql (insert)
+
+import GT.DB.Schema.Bike
+import GT.DB.Schema.Component
+import GT.DB.Schema.Usage
+import GT.DB.Schema.Ride
+
+-- | Create a new 'Usage' for all components on the 'Bike' for the given
+-- 'Ride'.
+--
+-- TODO: esqueleto does not support CTEs. It should!
```

Comment:
please [implement this](https://github.com/bitemyapp/esqueleto/issues/124), if only i knew any of the `esqueleto` maintainers..